src/app/modules/item/pages/item-edit/item-edit.component.ts

```diff
@@ -79,25 +81,54 @@ export class ItemEditComponent implements OnDestroy {
   }
 
   updateItemChanges(children: ChildData[]): void {
-    // FIXME: temp fix to compile and not send bad data to the service
-    this.itemChanges.children = children
-      .filter(child => child.id)
-      .map((child, idx) => ({ item_id: child.id as string, order: idx }));
+    this.itemChanges.children = children;
   }
 
+  // Update Item
+  private getItemChanges(children: ChildDataWithId[]): ItemChanges {
+    const res: ItemChanges = {};
+    if (this.itemChanges.children) res.children = children.map((child, idx) => ({ item_id: child.id, order: idx }));
```

Comment:
You loose the info that the id is not optional with this transform. 
"this.itemChanges" should not be used anymore around here as you give this children which is more complete.

Reply:
The id in ItemChanges isn't optional anymore, and I've removed that condition.